### PR TITLE
Change job functions to accept option object with backward compatability

### PR DESCRIPTION
**Pull request by std4lqi**
New UT is added to make sure this option object parameter will not break the existing string parameters.

### CODE REVIEW

File: lib/zosAccessor.js [hunk 1]
```diff
@@ -692,7 +787,7 @@ ZosAccessor.prototype.getJobLog = function (jobName, jobId, logFormat) {
                 });
                 readStream.on('end', function () {
                     var logString = Buffer.concat(chunks).toString('utf8');
-                    if (logFormat === 'metaInfo') {  // Deprecated
+                    if (fileId === 'metaInfo') {  // Deprecated
```
Comment (jiangg91) [inline lib/zosAccessor.js]: Altough it is deprecated, It should be`option.fileId` instead of `fileId`, right?

File: lib/zosAccessor.js [hunk 2]
```diff
@@ -411,42 +411,59 @@ ZosAccessor.prototype.rename = function (oldDataset, newDataset) {
...
         }
-        if (typeof condition === 'string') {
-            // Default condition is job name for compatibility.
-            condition = {jobName: condition};
-        }
         var siteParams = ['FILETYPE=JES'];
-        for(var key in condition) {
-            if (!condition.hasOwnProperty(key)) {
+        for(var key in option) {
+            if (!option.hasOwnProperty(key)) {
```
Comment (jiangg91) [inline lib/zosAccessor.js]: It is unnecessary to add this check `option.hasOwnProperty(key)`, right?

File: lib/zosAccessor.js (hunk 1 above)
Comment (std4lqi) [reply lib/zosAccessor.js]: The `option.fileId` is changed to `'x'`. Here we check the original fileId passed from parameter. We'll drop `'metaInfo'` in 2.0.0.

File: lib/zosAccessor.js (hunk 2 above)
Comment (std4lqi) [reply lib/zosAccessor.js]: `option.hasOwnProperty(key)` looks unnecessary here. It's kind of best practice to check whether `key` is owned by `option` object or inherited.

**Pull request merged**